### PR TITLE
Feat: new custom role resource

### PR DESCRIPTION
**Pull request by TomerHeber**
### Solution

Part of #498 
Implements client calls for roles + unit tests.

### CODE REVIEW

File: client/team_project_assignment.go [hunk 1]
```diff
@@ -4,26 +4,26 @@ import (
 	"errors"
 )
 
-type Role string
+type ProjectRole string
```
Comment (TomerHeber) [inline client/team_project_assignment.go:7]: name change to avoid conflicts...

File: client/role.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
+package client
+
+type RoleCreatePayload struct {
+	Name           string   `json:"name"`
+	OrganizationId string   `json:"organizationId"`
+	Permissions    []string `json:"permissions"`
+	IsDefaultRole  bool     `json:"isDefaultRole"`
+}
+
+type RoleUpdatePayload struct {
```
Comment (GiliFaroEnv0) [inline client/role.go:10]: you can use this as a base to other types

**Review by GiliFaroEnv0 — APPROVED** [review]
Looks great!
left a suggestion to reuse in the types

File: client/role.go (hunk 2 above)
Comment (TomerHeber) [reply client/role.go:10]: @GiliFaroEnv0 - it's not as clean in golang to do.
I'll see if I can improve it a little in the next PR. (Will merge this and review in the next PR).

**Comment by GiliFaroEnv0** [comment]
[Terraform Provider - Open a GH issue](https://trello.com/c/kubzY8fm/7306-terraform-provider-open-a-gh-issue)

**Pull request merged**